packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,522 @@
+    this.resizeObserver?.observe(this.el);
+  }
+
+  disconnectedCallback(): void {
+    disconnectLocalized(this);
+    disconnectMessages(this);
+    this.resizeObserver?.disconnect();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+  resizeObserver = createObserver("resize", () => this.resizeHandler());
+
+  private meterContainerEl: HTMLDivElement;
+
+  private valueLabelEl: HTMLDivElement;
+
+  private minLabelEl: HTMLDivElement;
+
+  private lowLabelEl: HTMLDivElement;
+
+  private highLabelEl: HTMLDivElement;
+
+  private maxLabelEl: HTMLDivElement;
+
+  private labelFlipMax = 0.8;
+
+  private labelFlipProximity = 0.15;
+
+  private minPercent = 0;
+
+  private maxPercent = 100;
+
+  @State() lowPercent: number;
+
+  @State() highPercent: number;
+
+  @State() lowActive: boolean;
+
+  @State() highActive: boolean;
+
+  @State() currentPercent: number;
+
+  @State() effectiveLocale: string;
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @State() defaultMessages: MeterMessages;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+  resizeHandler = (): void => {
+    this.updateLabels();
+  };
+
+  private updateLabels = (): void => {
+    if (this.valueLabelEl) {
+      this.determineValueLabelPosition();
+    }
+    if (this.rangeLabels) {
+      this.determineVisibleLabels();
+    }
+  };
+
+  private calculateValues() {
+    const { low, high, min, max, value } = this;
+    const lowPercent = (100 * (low - min)) / (max - min);
+    const highPercent = (100 * (high - min)) / (max - min);
+    const currentPercent = (100 * (value - min)) / (max - min);
+    this.lowPercent = lowPercent;
+    this.highPercent = highPercent;
+    this.currentPercent = value ? currentPercent : 0;
+    this.lowActive = !!low && low > min && (!value || low > value) && (!high || low < high);
+    this.highActive = !!high && high < max && (!value || high > value) && (!low || high > low);
+  }
+
+  /**
+   * Returns a string representing the localized label.
+   *
+   * @param value
+   * @param type
+   * @param percent
+   * @returns
+   */
+
+  private formatLabel = (value: number, labelType: MeterLabelType): string => {
+    if (typeof value === "number" && labelType !== "percent") {
+      numberStringFormatter.numberFormatOptions = {
+        locale: this.effectiveLocale,
+        numberingSystem: this.numberingSystem,
+        useGrouping: this.groupSeparator,
+      };
+      return numberStringFormatter.localize(value.toString());
+    } else if (typeof value === "number" && labelType === "percent") {
+      return Intl.NumberFormat(this.effectiveLocale, {
```

Comment:
Yet one more thing™ 😅, this may be called multiple times, so it'd be good to create and cache the formatter and reuse.

Reply:
Updated